packages/transaction-controller/src/TransactionController.ts

```diff
@@ -276,7 +276,7 @@ export class TransactionController extends BaseControllerV1<
 
   private readonly mutex = new Mutex();
 
-  private readonly nonceMutexByChainId = new Map<Hex, Mutex>();
+  private readonly nonceMutexesByChainId = new Map<Hex, Map<string, Mutex>>();
```

Comment:
alternatively to a nested map, we could use a flat map and construct the key of the map with some delimiter like `${chainId}-${key}`.

Reply:
I'm fine with either way! I think as is works!